### PR TITLE
Allow drivers and operators to update their own profiles

### PR DESCRIPTION
**Pull request by Cyebukayire**
### **Description**

This pull request implements the update profile feature for drivers and operators. 

### **How to test**
Run `npm install ` to install the dependencies
Run `npm run db:reseed` to use default registered users
Run `npm run dev` to start the development server
Navigate to http://localhost:{PORT}/api-docs for swagger documentation and head to Profile on `/api/profile/` endpoint.
Enter the body data using format of the default example provided 

Remember to use `1` as your `id` parameter for a successful profile update (Since the login feature is not yet merged to the develop branch we are hard coding a logged in use to have `user id = 1`)

### **Screenshot**
image

**Result**
image

### **Checklist**

- [x] The operator and driver are able to update information about their own profile.
- [x] Appropriate error messages are displayed on unsuccessful attempts.
- [x] Created `src/utils/responseHandler` to help handle success and failure responses without repeating the same codes for every API. 

### Note!
Very necessary issues that occurred on the database side (Couldn't run `db:migrate` and anything else related to migrations) were resolved by @PatrickNiyogitare28 on this [PR](https://github.com/atlp-rwanda/rca-phantom-team1-bn/pull/10). So I had to pull his work to be able to use the database even though the PR is not yet merged.

 ### **Changes Required**

- Since this feature is depending on the login feature that is not yet merged, I had to hard-code logged in user in `src/api/controller/profile.controller.js` which will be changed latter.
- The routes also are not protected; a middle ware to check athaurization and verify the user is expected later which is not part of this PR.

### CODE REVIEW

**Review by thepatrickniyo — COMMENTED** [review]
The feature looks good to me but I think you need to write tests for it.

**Comment by Cyebukayire** [comment]
> The feature looks good to me but I think you need to write tests for it.

Ohhh thank you @PatrickNiyogitare28, I should do that too

**Comment by thepatrickniyo** [comment]
> > The feature looks good to me but I think you need to write tests for it.
> 
> Ohhh thank you @PatrickNiyogitare28, I should do that too

Let me know if you need an assistance with it.

**Review by ernesteNtezirizaza — APPROVED** [review]

**Review by thepatrickniyo — APPROVED** [review]

File: src/api/routes/profile.routes.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/Profile'
+ *      404:
+ *        description: The profile was not found
+ *      500:
+ *        description: Some error happened
+ */
+
+profileRouter.get("/:id", userProfileExists, getProfile);
```
Comment (TuyizeeAnastase) [inline src/api/routes/profile.routes.js]: get("/profile", checkIfUSerLogged,getProfile)

File: src/api/routes/profile.routes.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+profileRouter.put(
+  "/:id",
+  checkUserLoggedIn,
+  restrictTo(ERoles.OPERATOR, ERoles.DRIVER),
+  userProfileExists,
+  allowedToEditProfile,
+  validateUpdateUserPayload,
+  updateAProfile
+);
```
Comment (TuyizeeAnastase) [inline src/api/routes/profile.routes.js]: put("/profile",checkUSerLogged,validation,updateProfile)

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]

**Pull request merged**